core/common/src/main/java/alluxio/util/CommonUtils.java

```diff
@@ -245,6 +245,30 @@ public static void waitFor(String description, Function<Void, Boolean> condition
   }
 
   /**
+   * Waits for an operation to return a non-null value with a specified timeout.
+   *
+   * @param description the description of this operation
+   * @param operation the operation
+   * @param options the options to use
+   * @param <T> the type of the return value
+   * @return the return value, null if it times out
+   */
+  public static <T> T waitForResult(String description, Function<Void, T> operation,
```

Comment:
yes, this is the true for other waitFor functions. It is just a convenient method for polling.  What do you suggest?